index.js

```diff
@@ -1,7 +1,8 @@
 var fs = require('fs')
+var buildMatcher = require('./may-contain').buildMatcher;
 
 module.exports = walkSync
-function walkSync (baseDir, relativePath) {
+function walkSync (baseDir, relativePath, matcher) {
```

Comment:
`relativePath` is only used internally, so we probably want `walkSync(baseDir, matcher, _relativePath)` as signature. (Maybe `globArray` or something like that instead of `matcher`?)